### PR TITLE
Variables

### PR DESCRIPTION
**Pull request by lenchen1112**
Translation for `1-js/02-first-steps/04-variables`.

### CODE REVIEW

File: 1-js/02-first-steps/04-variables/2-declare-variables/solution.md [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
 
 ```js
 let currentUserName = "John";
 ```
 
-Again, we could shorten that to `userName` if we know for sure that the user is current.
+同樣地，若確定 user 就代表現在的使用者，我們也可用 `userName` 讓它變得更短。
 
-Modern editors and autocomplete make long variable names easy to write. Don't save on them. A name with 3 words in it is fine.
+現代化編輯器的自動完成功能讓寫長命名變數變得很簡單，不要忽略這功能，一個名字有三個字節是適當的。
```
Comment (kaddopur) [inline 1-js/02-first-steps/04-variables/2-declare-variables/solution.md]:
```suggestion
現代化編輯器的自動完成功能讓寫長命名變數變得很簡單，不要忽略這功能，一個變數名中含有三個單字是適當的。
```

File: 1-js/02-first-steps/04-variables/2-declare-variables/solution.md [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+
+若你的編輯器沒有自動完成功能，快來選 [一個新的](/code-editors) 吧。
```
Comment (kaddopur) [inline 1-js/02-first-steps/04-variables/2-declare-variables/solution.md]:
```suggestion
若你的編輯器沒有自動完成功能，快來選[一個新的](/code-editors) 吧。
```

File: 1-js/02-first-steps/04-variables/3-uppercast-constant/solution.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
-We generally use upper case for constants that are "hard-coded". Or, in other words, when the value is known prior to execution and directly written into the code.
+我們通常使用大寫來命名 "寫死的" 常數，或者換句話說，當常數的值在程式執行前就已知並寫在程式碼內時。
 
-In this code, `birthday` is exactly like that. So we could use the upper case for it.
+這份程式碼中的 `birthday` 是這樣，所以我們可以讓它使用大寫命名。
+
+相對地，`age` 在執行期間才被計算出來。今天我們有一個年齡，一年後就會有另一個。某種程度上它因為不會經由程式碼被改變所以是個常數，但相較於 `birthday` 卻 "沒那麼不變"，它是被計算出來的，故我們應該用小寫來命名。
```
Comment (kaddopur) [inline 1-js/02-first-steps/04-variables/3-uppercast-constant/solution.md]:
```suggestion
相對地，`age` 在執行期間才被計算出來。今天我們有一個年齡，一年後就會有另一個。它在整個程式碼被執行的過程中不會改變，但相較於 `birthday` 卻 "沒那麼不變"，它是被計算出來的，故我們應該用小寫來命名。
```

File: 1-js/02-first-steps/04-variables/2-declare-variables/solution.md (hunk 2 above)
Comment (lenchen1112) [reply 1-js/02-first-steps/04-variables/2-declare-variables/solution.md]: 我這種引述前後都會留空白耶

Comment (kaddopur) [reply 1-js/02-first-steps/04-variables/2-declare-variables/solution.md]: 喔喔 我以為中文之間可以不用留空

Comment (kaddopur) [reply 1-js/02-first-steps/04-variables/2-declare-variables/solution.md]: 如果是強調用的話 我覺得ok

**Comment by lenchen1112** [comment]
@kaddopur 
check it out if there is anything else needs to be modified.

**Pull request merged**